src/lib.rs

```diff
@@ -777,6 +783,141 @@ impl Client {
         self.0.shutdown()
     }
 
+    /// Sets the x, y, width, and height properties of the current window. All values must be `>= 0`.
+    pub fn set_window_rect(&self, x: i32, y: i32, width: i32, height: i32) -> impl Future<Item = Self, Error = error::CmdError> + 'static {
+        use webdriver::common::Nullable;
+
+        debug_assert!(x >= 0);
+        debug_assert!(y >= 0);
+        debug_assert!(width >= 0);
+        debug_assert!(height >= 0);
+
+        let cmd = WebDriverCommand::SetWindowRect(webdriver::command::WindowRectParameters {
+            x: Nullable::Value(x),
+            y: Nullable::Value(y),
+            width: Nullable::Value(width),
+            height: Nullable::Value(height),
+        });
+
+        self.dup()
+            .issue_wd_cmd(cmd)
+            .map(|(this, _)| this)
+    }
+
+    /// Gets the x, y, width, and height properties of the current window.
+    pub fn get_window_rect(&self) -> impl Future<Item = (i32, i32, i32, i32), Error = error::CmdError> + 'static {
+        self.dup()
+            .issue_wd_cmd(WebDriverCommand::GetWindowRect)
+            .and_then(|(_, v)| match v {
+                Json::Object(obj) => {
+                    let x = match obj.get("x") {
+                        Some(Json::U64(v)) => *v as i32,
```

Comment:
Are these always `U64`? Is there not a generic "extract number" method on `Json` we can use?

Reply:
No the `Json` enum has helper methods to determine which type a value is but there is no generic "number" that matches the JS spec.

In practice these can/should be `F64` but in reality I'm always seeing `U64` values at this point in the code.

Reply:
I wonder if the other case should return a `NotW3C` error then?

Reply:
I tried that but `json` is partially-borrowed here.

Perhaps I need to construct a new `Json::Object` here.

I'll investigate tomorrow.

Reply:
If you use `obj.remove` instead of `obj.get`, it should work. You can also use [`Json::as_i64`](https://docs.rs/rustc-serialize/0.3.24/rustc_serialize/json/enum.Json.html#method.as_u64) for the casting.